l1-contracts/contracts/state-transition/chain-deps/facets/Getters.sol

```diff
@@ -36,8 +36,20 @@ contract GettersFacet is ZKChainBase, IGetters, ILegacyGetters {
     //////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IGetters
-    function getVerifier() external view returns (address) {
-        return address(s.verifier);
+    function getDualVerifier() external view returns (address) {
+        return address(s.dualVerifier);
+    }
+    /// @inheritdoc IGetters
+    function getPlonkVerifier() external view returns (address) {
+        return s.plonkVerifier;
+    }
+    /// @inheritdoc IGetters
+    function getFflonkVerifier() external view returns (address) {
+        return s.fflonkVerifier;
+    }
+    /// @inheritdoc IGetters
+    function getFflonkProofLength() external view returns (uint256) {
+        return s.fflonkProofLength;
     }
```

Comment:
ZK chain should know nothing about fflonk/plonk or dual verifier. It has only one contract - verifier and it is an abstraction for the contract what proof system is inside